samples/nrf_desktop/src/hw_interface/buttons.c

```diff
@@ -81,14 +86,17 @@ static int set_trig_mode(int trig_mode)
 	__ASSERT_NO_MSG((trig_mode == GPIO_INT_EDGE) ||
 			(trig_mode == GPIO_INT_LEVEL));
 
-	int flags = GPIO_PUD_PULL_DOWN | GPIO_DIR_IN | GPIO_INT |
-		    GPIO_INT_ACTIVE_HIGH;
-	flags |= trig_mode;
+	int flags = (IS_ENABLED(CONFIG_DESKTOP_BUTTONS_POLARITY_INVERSED) ?
+		(GPIO_PUD_PULL_UP | GPIO_INT_ACTIVE_LOW) :
+		(GPIO_PUD_PULL_DOWN | GPIO_INT_ACTIVE_HIGH));
+	flags |= (GPIO_DIR_IN | GPIO_INT | trig_mode);
 
 	int err = 0;
 
-	for (size_t i = 0; (i < ARRAY_SIZE(row_pin)) && !err; i++) {
-		err = gpio_pin_configure(gpio_dev, row_pin[i], flags);
+	for (size_t i = 0; (i < ARRAY_SIZE(row)) && !err; i++) {
+		__ASSERT_NO_MSG(row[i].port < ARRAY_SIZE(port_map));
```

Comment:
assert is fine but please move it to initialization function where this port number is used for the first time

Reply:
`set_trig_mode()` is used for both initialization and reconfiguration. It's the first place where the port is used